### PR TITLE
lfo display caching

### PR DESCRIPTION
**Pull request by blancoberg**
* caches the lfo display
* only redraws when parameters are changed, lfo switched or lua code is applied

addresses partially #7895

### CODE REVIEW

File: src/surge-xt/gui/widgets/LFOAndStepDisplay.cpp [hunk 1]
```diff
@@ -414,12 +417,112 @@ void LFOAndStepDisplay::paint(juce::Graphics &g)
...
     }
 
     paintTypeSelector(g);
 }
 
+bool LFOAndStepDisplay::paramsHasChanged()
+{
+    bool hasChanged = false;
+
+    if (paramsFromLastDrawCall[lfodata->delay.param_id_in_scene].i != lfodata->delay.val.i)
```
Comment (baconpaul) [inline src/surge-xt/gui/widgets/LFOAndStepDisplay.cpp:455]: Happy to merge this, but the lfo data has params which are contiguous so you could also make this code

```
   auto *p = &lfodata->delay; // look in the definition of LFOStorage for which param is first
  while (p <= &lfodata->shape) // and last
   {
      if (paramsFromLastDrawCa[[p->paramid].i != p->val.i;
         hasChanged = true;
   }
   ```
   
   type thing. If you want to make this change now I can wait but I'm also happy to merge.

Thanks for doing this!

Comment (blancoberg) [reply src/surge-xt/gui/widgets/LFOAndStepDisplay.cpp:455]: That is much cleaner.  I was looking for a way of doing that but Couldn't figure out how.  I'll fix it

Comment (blancoberg) [reply src/surge-xt/gui/widgets/LFOAndStepDisplay.cpp:455]: image

Not sure how to interpret the order here. Does this mean that rate is first and release last? Or is there some other list that sets the order of these?

Comment (baconpaul) [reply src/surge-xt/gui/widgets/LFOAndStepDisplay.cpp:455]: rate is first and release is last yes.

**Comment by blancoberg** [comment]
nope those are not included. 
any hints on where I can catch those events?

**Pull request merged**